docs/adding_a_feature.md

```diff
@@ -0,0 +1,211 @@
+This is an overview of how we added a feature:
+
+To make auto-upgrading on the nodes an option.
+
+Auto-upgrades are configured by nodeup.  nodeup is driven by the nodeup model.
+
+Inside the nodeup model there are folders which serve three roles:
+
+1. A folder with a well-known name means that items under that folder are treated as items of that type:
```

Comment:
Numbering is monkeyed by Github markdown rendering.  I think it is a bug in there rendering.  It is rendering as 1. then 1. then 2.